contrib/node/src/python/pants/contrib/node/tasks/node_paths.py

```diff
@@ -41,3 +41,23 @@ def all_node_paths(self):
     :rtype list string
     """
     return list(self._paths_by_target.values())
+
+
+class NodePaths(NodePathsBase):
+  """Maps NpmPackage targets to their resolved NODE_PATH chroot.
+
+  NodePaths is exposed as a product type for resolve.node.
+  The paths are the resolved chroot after copying sources to the synthetic workspace.
+  """
+
+
+class NodePathsLocal(NodePathsBase):
+  """Maps Node package targets to their local NODE_PATH chroot.
```

Comment:
The "local" bit here is still not well defined. I _think_ that this means "outside of `.pants.d` and next to the target definition"? Except maybe for 3rdparty packages?

Reply:
The "local" bit is I think still not clear, as evidenced by Benjy's question.

Maybe you could have some/all of these docstrings refer to the docstring of `NodeResolve`, which you expanded?